ui/pages/confirm-decrypt-message/confirm-decrypt-message.component.js

```diff
@@ -13,118 +14,80 @@ import { MetaMetricsEventCategory } from '../../../shared/constants/metametrics'
 import { SECOND } from '../../../shared/constants/time';
 import { Numeric } from '../../../shared/modules/Numeric';
 import { EtherDenomination } from '../../../shared/constants/common';
-import { Icon, IconName } from '../../components/component-library';
-import { IconColor } from '../../helpers/constants/design-system';
+import {
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+  Icon,
+} from '../../components/component-library';
 import { formatCurrency } from '../../helpers/utils/confirm-tx.util';
 import { getValueFromWeiHex } from '../../../shared/modules/conversion.utils';
 import { COPY_OPTIONS } from '../../../shared/constants/copy';
+import { useI18nContext } from '../../hooks/useI18nContext';
+import { useScrollRequired } from '../../hooks/useScrollRequired';
+import { MetaMetricsContext } from '../../contexts/metametrics';
+import {
+  BackgroundColor,
+  BorderRadius,
+  Display,
+  IconColor,
+} from '../../helpers/constants/design-system';
 
-export default class ConfirmDecryptMessage extends Component {
-  static contextTypes = {
-    t: PropTypes.func.isRequired,
-    trackEvent: PropTypes.func.isRequired,
-  };
-
-  static propTypes = {
-    fromAccount: PropTypes.shape({
-      address: PropTypes.string.isRequired,
-      balance: PropTypes.string,
-      name: PropTypes.string,
-    }).isRequired,
-    clearConfirmTransaction: PropTypes.func.isRequired,
-    cancelDecryptMessage: PropTypes.func.isRequired,
-    decryptMessage: PropTypes.func.isRequired,
-    decryptMessageInline: PropTypes.func.isRequired,
-    history: PropTypes.object.isRequired,
-    mostRecentOverviewPage: PropTypes.string.isRequired,
-    requesterAddress: PropTypes.string,
-    txData: PropTypes.object,
-    subjectMetadata: PropTypes.object,
-    nativeCurrency: PropTypes.string.isRequired,
-    currentCurrency: PropTypes.string.isRequired,
-    conversionRate: PropTypes.number,
-  };
-
-  state = {
-    copyToClipboardPressed: false,
-    hasCopied: false,
-  };
-
-  copyMessage = () => {
-    copyToClipboard(this.state.rawMessage, COPY_OPTIONS);
-    this.context.trackEvent({
-      category: MetaMetricsEventCategory.Messages,
-      event: 'Copy',
-      properties: {
-        action: 'Decrypt Message Copy',
-        legacy_event: true,
-      },
-    });
-    this.setState({ hasCopied: true });
-    setTimeout(() => this.setState({ hasCopied: false }), SECOND * 3);
-  };
-
-  renderHeader = () => {
-    return (
-      <div className="request-decrypt-message__header">
-        <div className="request-decrypt-message__header-background" />
-
-        <div className="request-decrypt-message__header__text">
-          {this.context.t('decryptRequest')}
-        </div>
-
-        <div className="request-decrypt-message__header__tip-container">
-          <div className="request-decrypt-message__header__tip" />
-        </div>
+const Header = () => {
```

Comment:
Given we're simultaneously doing these refactors, should we highlight the new visibility logic with a PR comment?

Reply:
Do you mean the visibility of scroll and the scrollToBottom button? It's already noted in PR description